pkg/metrics/statsd/statsd_test.go

```diff
@@ -83,3 +86,247 @@ func TestEmitFloat(t *testing.T) {
 		t.Error(m)
 	}
 }
+
+func TestEmitGaugeViaUDP(t *testing.T) {
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+
+	env := mock_env.NewMockInterface(controller)
+	env.EXPECT().Location().AnyTimes().Return("eastus")
+	env.EXPECT().Hostname().AnyTimes().Return("test-host")
+
+	socket := "127.0.0.1:8001"
+
+	os.Setenv(statsdSocketEnv, "udp:"+socket)
```

Comment:
IMO - env manipulations in tests is usually a bad idea. This can lead to flakes: we set env in this test, but we might rely on this env in a different test. Depending on the order of execution we might see failure/success.

I think we can solve this by removing `os.LookupEnv` call from `getConnectionDetails()` and passing a value into `New` similar to `account` and `namespace`.

UPD: In a different comment I say that we can drop this test, but I still think we should remove `os.LookupEnv` from `statsd`.